src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -72,6 +88,18 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                     // Find supported, but not yet searchable params
                     var notYetIndexedParams = _supportedSearchParameterDefinitionManager.GetSupportedButNotSearchableParams();
 
+                    // if there are not any parameters which are supported but not yet indexed, then we have nothing to do
+                    if (!notYetIndexedParams.Any())
+                    {
+                        _reindexJobRecord.Error.Add(new OperationOutcomeIssue(
+                            OperationOutcomeConstants.IssueSeverity.Information,
+                            OperationOutcomeConstants.IssueType.Informational,
+                            Resources.NoSearchParametersNeededToBeIndexed));
+                        _reindexJobRecord.CanceledTime = DateTimeOffset.UtcNow;
+                        await CompleteJobAsync(OperationStatus.Canceled, cancellationToken);
```

Comment:
This seems like it would be a completed outcome. There were no parameters to reindex, so all of the reindexing got done.

Reply:
I marked it as canceled because no changes occurred in the server.  We did not update the hash value on any resources, or touch anything else.  I do see your point that there were no errors and it that sense it completed by checking if any work needed to be done.  I am open to changing it to complete.